pytket/qir/conversion/conversion.py

```diff
@@ -855,9 +847,47 @@ def condition_block() -> None:
                     )
 
             elif isinstance(op, SetBitsOp):
-                _, outputs = self._get_c_regs_from_com(command)
-                for out in outputs:
-                    self.set_cregs[out] = command.op.values
+                assert len(command.op.values) == len(command.bits)
+                assert len(command.qubits) == 0
+
+                for b, v in zip(command.bits, command.op.values):
+                    if v:
+                        output_instruction = pyqir.const(self.qir_bool_type, 1)
+                    else:
+                        output_instruction = pyqir.const(self.qir_bool_type, 0)
+
+                    self.module.builder.call(
+                        self.set_one_bit_in_reg,
+                        [
+                            self.ssa_vars[b.reg_name],
+                            pyqir.const(self.qir_int_type, b.index[0]),
+                            output_instruction,
+                        ],
+                    )
+
+            elif isinstance(op, CopyBitsOp):
+                assert len(command.qubits) == 0
+                half_length = int(len(command.args) / 2)
+                assert 2 * half_length == len(command.args)
```

Comment:
```suggestion
                assert len(command.args) % 2 == 0
                half_length = len(command.args) // 2
```
Equivalent but clearer imo.

Reply:
Done in @cqc-melf
[suggested changes: CopyBits](https://github.com/CQCL/pytket-qir/pull/45/commits/de8d6e3272fa6ae482bcf28f5753a8087a55ddf8)
[de8d6e3](https://github.com/CQCL/pytket-qir/pull/45/commits/de8d6e3272fa6ae482bcf28f5753a8087a55ddf8)